src/main/java/core/basesyntax/operation/BalanceHandler.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.service.ShopService;
+import core.basesyntax.transaction.FruitTransaction;
+
+public class BalanceHandler implements OperationHandler {

```

Comment:
According to the SOLID principles, your class should focus on a single problem. In this case, it's better to remove the check for the operation type from the `apply` method and just assume that the correct operation will be passed. This way, your `BalanceHandler` will only be responsible for handling the 'balance' operation, which makes it more focused and easier to maintain.